presto-main/src/main/java/io/prestosql/execution/SqlTaskExecution.java

```diff
@@ -641,6 +641,7 @@ private synchronized void checkTaskCompletion()
 
         // are there still pages in the output buffer
         if (!outputBuffer.isFinished()) {
```

Comment:
@dain is this a valid transition for grouped execution or there should be additional conditions? I'm thinking that if there are more execution groups, the task should stay in `RUNNING` state.